pression/data/Compressor.h

```diff
@@ -0,0 +1,126 @@
+
+/* Copyright (c) 2016, Stefan.Eilemann@epfl.ch
+ *
+ * This file is part of Pression <https://github.com/Eyescale/Pression>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
+
+#include <pression/data/api.h>
+#include <pression/data/types.h>
+#include <lunchbox/buffer.h> // used inline
+#include <lunchbox/compiler.h> // LB_UNUSED
+#include <lunchbox/debug.h> // LBUNIMPLEMENTED
+
+namespace pression
+{
+namespace data
+{
+/** Base interface for lossless CPU compressors of binary data. */
+class Compressor
+{
+public:
+    virtual ~Compressor() {} //!< @internal
+
+    typedef lunchbox::Bufferb Result; //!< Single result data buffer
+    typedef std::vector< Result > Results; //!< Set of result chunks
+
+    /**
+     * Compress the given data and return the result.
+     *
+     * This default implementation will use getCompressBound(), getChunkSize()
+     * and call the protected compress() method to parallelize the
+     * compression. The result is valid until the next call to compress or
+     * destruction of this instance.
```

Comment:
I think you should mention that this class doesn't produce an output which is suitable for long term in memory compression (whether it makes sense aside) as it getCompressBound can indeed return a size larger than the original data.

Aso, this comment should go to the protected part of the interface (not sure where exactly) as it's a detail more suitable for users of the protected API than the public. The client may be interested in the fact that in runs in parallalel, but nothing more. Another option is to move the requirements on derived classes to the class documentation, which is really the documentation entry point.